code/modules/power/apc/apc_attack.dm

```diff
@@ -264,6 +264,9 @@
 	if(machine_stat & BROKEN && obj_integrity <= 0)
 		if(sound_effect)
 			play_attack_sound(damage_amount, damage_type, damage_flag)
+		new /obj/item/stock_parts/cell/upgraded/empty(loc)
```

Comment:
Have it drop the current cell inside instead of a generic cell.

Reply:
Tried it, it gets qde'dl with the apc not sure how to remove it fully, I did forceMove to loc previously so I need to clear it from the apc before it qdel

Reply:
Fixed it